src/components/encoding-pane/property-editor-schema.ts

```diff
@@ -0,0 +1,96 @@
+import * as vlSchema from 'vega-lite/build/vega-lite-schema.json';
+
+export interface PropertyEditorFormatSchema {
+  type: string;
+  title: string;
+  properties: any;
+}
+
+export interface PropertyEditorSchema {
+  uiSchema: any;
+  schema: PropertyEditorFormatSchema;
+}
+
+export function getPropertyEditorSchema(prop: string, nestedProp: string): PropertyEditorSchema {
+  if (prop === 'scale') {
+    if (nestedProp === 'type') {
+      return generateSchema(prop, nestedProp, SCALE_TYPE_UISCHEMA,
+        (vlSchema as any).definitions.ScaleType.enum);
+    }
+  } else if (prop === 'axis') {
+    if (nestedProp === 'orient') {
+      return generateSchema(prop, nestedProp, AXIS_ORIENT_UISCHEMA, (vlSchema as any).definitions.AxisOrient.enum);
+    } else if (nestedProp === 'title') {
+      return generateSchema(prop, nestedProp, AXIS_TITLE_UISCHEMA);
+    }
+  } else if (prop === 'stack') {
+    return generateSchema(prop, nestedProp, STACK_UISCHEMA, (vlSchema as any).definitions.StackOffset.enum);
+  } else {
+    return {
+      schema: undefined,
+      uiSchema: {}
+    };
+  }
+}
+
+// Capitalize first letter for aesthetic purposes in form
+function capitalizeFirstLetter(s: string): string {
+  return s.charAt(0).toUpperCase() + s.slice(1);
+}
+
+/*
+*  NOTE: factory method where propertyKey follows naming convention: Capitalized Prop + Capitalized NestedProp
+*  Example: {prop: axis, nestedProp: orient} translates to "AxisOrient"
+*/
+function generateSchema(prop: string, nestedProp: string, uiSchema: any, schemaEnum?: any): PropertyEditorSchema {
+  prop = capitalizeFirstLetter(prop);
+  nestedProp = nestedProp ? capitalizeFirstLetter(nestedProp) : nestedProp;
```

Comment:
We agreed to use `_`, not camelCase nor PascalCase.  

https://github.com/vega/voyager/pull/789#discussion_r183281143